src/DelegateDecompiler/Processor.cs

```diff
@@ -998,8 +1012,14 @@ static void StElem(ProcessorState state)
             if (newArray != null)
             {
                 var expressions = CreateArrayInitExpressions(newArray, value, index);
-                var newArrayInit = Expression.NewArrayInit(array.Type.GetElementType(), expressions);
-                array.Expression = newArrayInit;
+                if (typeof(ParameterExpression).IsAssignableFrom(array.Type.GetElementType()))
+                {
+                    array.Expression = Expression.Constant(expressions);

```

Comment:
Not sure about this line since I encountered only ParameterExpressinon[1] occurrences